app/controllers/stories_controller.rb

```diff
@@ -164,7 +164,7 @@ def handle_user_index
       return
     end
     assign_user_comments
-    @pinned_stories = Article.published.where(id: @user.profile_pins.pluck(:pinnable_id)).
+    @pinned_stories = Article.published.where(id: @user.profile_pins.select(:pinnable_id)).
```

Comment:
This will be `SELECT articles.* FROM articles where published = true AND id IN (select pinnable_id from profile_pins) etc. etc`